pkg/scheduler/extender_test.go

```diff
@@ -277,12 +277,15 @@ func TestGenericSchedulerWithExtenders(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			scheduler := NewGenericScheduler(
-				cache,
-				emptySnapshot,
-				schedulerapi.DefaultPercentageOfNodesToScore)
+			scheduler := &Scheduler{
+				SchedulerCache:           cache,
+				Schedule:                 Schedule,
+				Extenders:                extenders,
+				nodeInfoSnapshot:         emptySnapshot,
+				percentageOfNodesToScore: schedulerapi.DefaultPercentageOfNodesToScore,
+			}
 			podIgnored := &v1.Pod{}
-			result, err := scheduler.Schedule(context.Background(), extenders, fwk, framework.NewCycleState(), podIgnored)
+			result, err := scheduler.Schedule(context.Background(), scheduler, fwk, framework.NewCycleState(), podIgnored)
```

Comment:
it's weird to for `scheduler` calling `Schedule()` and pass itself as a parameter, so I thought we still need a interface, and implement it with our default scheduler(for now it's genericScheduler), I think it's more extensible. But I found @alculquicondor has already discussed about getting entirely rid of `genericScheduler`, glad to hear some advices from alculquicondor.

Changing the input may be another alternative way.

Reply:
Indeed. It's a weird.
The `Schedule()` is implemented to a mock function in `scheduler_test.go/TestSchedulerScheduleOne()`, I didn't find a better solution for now.
I think, rewrite the `TestSchedulerScheduleOne` and make `Schedule()` a method of `Scheduler` may be a better way.

Let's hear from opinions from alculquicondor.

Reply:
How about use `sched` as you used in the following code? Or we keep `extender`? Naming is difficult...

Actually is it possible to remove `extenders []framework.Extender` in the function parameter since it could call itself inside the implementation now.

```diff
diff --git a/pkg/scheduler/generic_scheduler.go b/pkg/scheduler/generic_scheduler.go
index 5d7a6c69442..19a79eeb204 100644
--- a/pkg/scheduler/generic_scheduler.go
+++ b/pkg/scheduler/generic_scheduler.go
@@ -90,7 +90,7 @@ func (g *genericScheduler) snapshot() error {
 // Schedule tries to schedule the given pod to one of the nodes in the node list.
 // If it succeeds, it will return the name of the node.
 // If it fails, it will return a FitError error with reasons.
-func (g *genericScheduler) Schedule(ctx context.Context, extenders []framework.Extender, fwk framework.Framework, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
+func (g *genericScheduler) Schedule(ctx context.Context, fwk framework.Framework, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
        trace := utiltrace.New("Scheduling", utiltrace.Field{Key: "namespace", Value: pod.Namespace}, utiltrace.Field{Key: "name", Value: pod.Name})
        defer trace.LogIfLong(100 * time.Millisecond)

@@ -103,7 +103,7 @@ func (g *genericScheduler) Schedule(ctx context.Context, extenders []framework.E
                return result, ErrNoNodesAvailable
        }

-       feasibleNodes, diagnosis, err := g.findNodesThatFitPod(ctx, extenders, fwk, state, pod)
+       feasibleNodes, diagnosis, err := g.findNodesThatFitPod(ctx, fwk, state, pod)
        if err != nil {
                return result, err
        }
@@ -126,7 +126,7 @@ func (g *genericScheduler) Schedule(ctx context.Context, extenders []framework.E
                }, nil
        }

-       priorityList, err := prioritizeNodes(ctx, extenders, fwk, state, pod, feasibleNodes)
+       priorityList, err := prioritizeNodes(ctx, g, fwk, state, pod, feasibleNodes)
        if err != nil {
                return result, err
        }
```

Reply:
I didn't get what you mean, could you explain more details? 😃

Reply:
Ohoh, @ruquanzhao I mean if it is possible to remove the extenders parameter in the interface declaration.

Reply:
we don't have the interface anymore, it has been removed in this PR.